fs/fs.go

```diff
@@ -74,20 +74,67 @@ import (
 	"github.com/opencontainers/go-digest"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
 	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
 )
 
 var (
 	defaultIndexSelectionPolicy = SelectFirstPolicy
 	fusermountBin               = "fusermount"
+	layerQueueBufferSize        = 1024
 )
 
+type layerQueue struct {
+	queue chan func() string
+	cache *sync.Map
+	smp   *semaphore.Weighted
+}
+
+func newLayerQueue(maxConcurrency int64) *layerQueue {
+	lq := &layerQueue{}
+	lq.queue = make(chan func() string) // arbitrarily chosen buffer size
+	lq.cache = &sync.Map{}
+	lq.smp = semaphore.NewWeighted(maxConcurrency)
+	return nil
+}
+
+func (lq *layerQueue) Start(ctx context.Context) error {
```

Comment:
I think this errgroup and context stuff is done correctly, just want to double check. Using the same context as the filesystem ties it to the lifecycle of the filesystem, which I think makes sense since we only use this in fs.Mount.